### PR TITLE
[inductor] Support get_sizes and get_strides in C shim

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109436
* __->__ #109351
* #109022

Summary: These are needed for supporting dynamic shapes in AOTInductor. Also refactor tests with multiple inputs in test_aot_inductor.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

### CODE REVIEW

File: torch/csrc/inductor/aoti_torch/c/shim.h [hunk 1]
```diff
@@ -98,6 +98,9 @@ AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE int32_t aoti_torch_dtype_int16();
 AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE int32_t aoti_torch_dtype_int32();
 AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE int32_t aoti_torch_dtype_int64();
 
+AOTI_TORCH_EXPORT AOTI_TORCH_NOINLINE int64_t
```
Comment (desertfire) [inline torch/csrc/inductor/aoti_torch/c/shim.h:101]: @ezyang , I really don't want to return AOTITorchError for this one. I guess theoretically we can hit a div_by_zero exception, but I wonder if it is really worth to do that.

Comment (desertfire) [reply torch/csrc/inductor/aoti_torch/c/shim.h:101]: Maybe a better way is to duplicate https://github.com/pytorch/pytorch/blob/1cc0921eb62089392595264610cfa863d42fe9bc/aten/src/ATen/native/BinaryOps.h#L97 (without using c10 functions in it) in model.so for simple utility function like this?

Comment (ezyang) [reply torch/csrc/inductor/aoti_torch/c/shim.h:101]: uhhhh why are you sending this across the ABI boundary, why not just have an implementation in model.so

Comment (ezyang) [reply torch/csrc/inductor/aoti_torch/c/shim.h:101]: yeah what you suggested

**Review by chenyang78 — APPROVED** [review]
LGTM. Thanks!

File: torch/_inductor/codegen/wrapper.py [hunk 2]
```diff
@@ -965,10 +969,24 @@ def __init__(self):
...
-        from .cpp import cexpr
+        from .cpp import cexpr, CppPrinter
 
         self.expr_printer = cexpr
 
+        # CppPrinter sometimes calls at::native functions which causes problems in
+        # the ABI compatible mode. Currently we are hitting this problem when codegen
+        # Grid computation expressions, but we my need to fix other size computation
+        # as well.
+        class GridExprCppPrinter(CppPrinter):
```
Comment (ezyang) [inline torch/_inductor/codegen/wrapper.py:980]: Put this top level?

**Review by ezyang — APPROVED** [review]

**Comment by desertfire** [comment]
merged in another exported PR, https://github.com/pytorch/pytorch/pull/109391

**Pull request closed without merging**